packages/cozy-harvest-lib/src/components/TriggerLauncher.jsx

```diff
@@ -141,9 +151,13 @@ export class TriggerLauncher extends Component {
         })}
         {showTwoFAModal && (
           <TwoFAModal
-            konnectorJob={konnectorJob}
             dismissAction={this.dismissTwoFAModal}
             into="coz-harvest-modal-place"
+            konnectorSlug={triggersModel.getKonnectorSlug(trigger)}
+            hasFailed={this.konnectorJob.isTwoFARetry()}
+            isSubmitting={this.konnectorJob.isTwoFARunning()}
+            onTwoFACodeSubmit={this.handleTwoFACode}
+            provider={this.konnectorJob.getTwoFACodeProvider()}
```

Comment:
Why did we remove konnectorJob as prop and created new props ? It seems that by doing so, we introduce complexity and a new vocabulary that might not be necessary. If I look at the code of TwoFAModal I see `hasFailed` and I think what does `hasFailed` mean, and I must come here to see that `isTwoFaRetry` has been transformed into `hasFailed` ? Why introducing a layer of translation here ?

Reply:
Talking with Yannick, I understand that the need is to prevent TwoFAModal to call methods on konnectorJob. An alternative way would be to have a `.freeze()`, `.dumb()` method on konnectorJob that would pass a dumbed down version of konnectorJob (without the methods) to TwoFAModal.

My impression here and what I want to prevent is : I do not want, in the event of an evolution of konnectorJob or of the needs of TwoFAModal to have to change TriggerLauncher. I just want to chagne the code of TwoFAModal or konnectorJob.

Reply:
Well spotted @ptbrowne !

Actually the best solution is to pass the konnectorJob and subscribe to its own event mechanism. It is an evidence for us today but yesterday we were a little bit both tired at the end of the day.